benchmark/dqn/algorithm.py

```diff
@@ -0,0 +1,88 @@
+#   Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import warnings
+warnings.simplefilter('default')
+
+import copy
+import torch
+import torch.optim as optim
+import torch.nn.functional as F
+from torch.nn.utils import clip_grad_norm_
+from parl.core.torch.algorithm import Algorithm
+from parl.utils.deprecation import deprecated
+import numpy as np
+
+__all__ = ['DQN']
+
+
+class DQN(Algorithm):
+    def __init__(self, model, algo='DQN', act_dim=None, gamma=None, lr=None):
+        """ DQN algorithm
+        
+        Args:
+            model (parl.Model): model defining forward network of Q function.
+            algo (str): which dqn model to use.
```

Comment:
Can we have three separate algorithms of DQN variants?  We hope that the users import algorithms from parl.algorithm, such as `parl.algorithms.DQN`,  `parl.algorithms.DDQN`, or`parl.algorithms.DuelingDQN`.